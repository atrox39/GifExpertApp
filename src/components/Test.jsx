import PropTypes from 'prop-types';

function Test({ mensaje }) {
  return (
    <div>
      <h1>{mensaje}</h1>
    </div>
  );
}

Test.propTypes = {
  mensaje: PropTypes.string.isRequired,
};

export default Test;
