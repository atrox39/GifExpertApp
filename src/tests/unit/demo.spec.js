import React from 'react';
import { create } from 'react-test-renderer';
import Test from '../../components/Test';

describe('Demo Test', () => {
  const TEXT = 'Test';
  const component = create(<Test mensaje={TEXT} />);

  it('Simple test', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Debe tener un texto de la propiedad mensaje', async () => {
    const h1 = await component.root.findByType('h1');
    expect(h1.props.children).toEqual(TEXT);
  });
});
