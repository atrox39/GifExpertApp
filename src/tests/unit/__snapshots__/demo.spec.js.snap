// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo Test Simple test 1`] = `
<div>
  <h1>
    Test
  </h1>
</div>
`;
